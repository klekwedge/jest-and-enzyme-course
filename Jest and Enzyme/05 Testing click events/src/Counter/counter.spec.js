import React from "react";
import Counter from "./counter";

const setUp = () => shallow(<Counter />);

describe("Count component", () => {
  let component;
  let instance;
  beforeEach(() => {
    component = setUp();
    instance = component.instance();
  });

  it("should render Counter component", () => {
    expect(component).toMatchSnapshot();
  });

  describe("Counter handles", () => {
    it("should render value to plus one", () => {
      const btn = component.find(".plusOneBtn");
      btn.simulate("click");
      // expect(component).toMatchSnapshot();
      expect(component.state().count).toBe(1);
    });

    it("should render value to 10", () => {
      const btn = component.find(".resetBtn");
      btn.simulate("click");
      expect(component.state().count).toBe(10);
    });

    it("should render value to custom value", () => {
      const btn = component.find(".resetBtn");
      btn.simulate("click");
      instance.handleReset(20);
      expect(component.state().count).toBe(20);
    });
  });
});
