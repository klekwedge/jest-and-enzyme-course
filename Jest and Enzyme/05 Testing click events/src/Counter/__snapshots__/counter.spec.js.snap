// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count component Counter handles should render value to plus one 1`] = `
<div>
  <div>
    1
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;

exports[`Count component should render Counter component 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;
